casper/src/test/scala/coop/rchain/casper/ValidateTest.scala

```diff
@@ -281,6 +281,51 @@ class ValidateTest
       } yield ()
   }
 
+  "Future deploy validation" should "not accept blocks with a deploy for a future block number" in withStorage {
+    implicit blockStore => implicit blockDagStorage =>
+      for {
+        deploy <- ProtoUtil.basicProcessedDeploy[Task](0)
+        block <- createBlock[Task](
+                  Seq.empty[BlockHash],
+                  deploys = Seq(deploy)
+                )
+        status <- Validate.futureTransaction[Task](block)
+        _      = status should be(Right(Valid))
+
+        deployData        = deploy.deploy.get
+        updatedDeployData = deployData.withValidAfterBlockNumber(Long.MaxValue)
+        blockWithFutureDeploy <- createBlock[Task](
+                                  Seq.empty[BlockHash],
+                                  deploys = Seq(deploy.withDeploy(updatedDeployData))
+                                )
+        status2 <- Validate.futureTransaction[Task](blockWithFutureDeploy)
+        _       = status2 should be(Left(ContainsFutureDeploy))
+      } yield ()
+  }
```

Comment:
This test can be split in two: one for valid scenario and one for the invalid one